pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Consumer.java

```diff
@@ -255,6 +255,13 @@
      * @return a future that can be used to track the completion of the operation
      */
     CompletableFuture<Void> acknowledgeCumulativeAsync(MessageId messageId);
+    
+    /**
+     * Flush all batched acknowledgements and wait until all acknowledgments have been persisted.
+     * 
+     * Flush acks returns immediately if batching of acks is disabled.
+     */
+    void flushAcknowledgements();
```

Comment:
in Consumer api, we have both sync and async api. so I think it would be good to add flushAsync in acknowledgmentsGroupingTracker, convert `flush` to use `flushAsync, and then expose flushAsync as flushAcknowledgementsAsync() and expose flush as flushAcknowledgments

Reply:
Since there's no "ack on the ack", the sync vs async won't matter much. We had it initially on the acknowledge, though that was never a great idea.

Reply:
There is a difference here. Because it is a blocking call waiting for it to be written to the wire, no?

Reply:
It should be a blocking call, in my opinion, since it requires a channel flush on the client ctx which is a blocking call.

Reply:
yes. netty provide a callback for flush operation, right?

Reply:
It looks like netty's flush call starts a thread for running the flush action, but there is no callback for the client. So it is not a blocking call waiting for the data to be written to the wire.

Reply:
I believe you can get `cnx.ctx().flush().voidPromise()`, no?